app/components/ak-pagination-provider/index.ts

```diff
@@ -36,9 +36,11 @@ export interface AkPaginationProviderSignature<R> {
     totalItems: number;
     itemPerPageOptions: number[];
     defaultLimit: number;
+    tableParams?: string[];
```

Comment:
dependencies should allow more types
```
type ResultDependency = 
  | string                  // e.g., "search", "category", etc.
  | number                  // e.g., for pagination, limit, etc.
  | boolean                 // e.g., isActive, sortOrder, etc.
  | string[]                // e.g., tags, selected categories, etc.
  | number[]                // e.g., multiple selected IDs
  | null                    // Nullable params (e.g., optional filters)
```

`resultDependencies: ResultDependency[];`